internal/scout/usage/k8s.go

```diff
@@ -0,0 +1,384 @@
+	if err != nil {
+		return 0, errors.Wrap(err, "error while converting inf.Dec to float")
+	}
+
+	return toFloat, nil
+}
+
+// getPvcCapacity retrieves the storage capacity of a PersistentVolumeClaim
+// mounted as a volume by a container.
+//
+// It accepts:
+// - ctx: The context for the request
+// - cfg: A config struct containing:
+//   - k8sClient: A Kubernetes clientset for accessing the API server
+//   - namespace: The namespace of the pod
+// - container: The container object which may have PVC mounts
+// - pod: The pod object which contains the container
+//
+// It returns:
+// - The capacity Quantity of the PVC if a matching PVC mount is found
+// - nil if no PVC mount is found
+// - Any error that occurred while getting the PVC
+func getPvcCapacity(ctx context.Context, cfg *Config, container v1.Container, pod *v1.Pod) (*resource.Quantity, error) {
+	for _, volumeMount := range container.VolumeMounts {
+		for _, volume := range pod.Spec.Volumes {
+			if volume.Name == volumeMount.Name && volume.PersistentVolumeClaim != nil {
+				pvc, err := cfg.k8sClient.CoreV1().PersistentVolumeClaims(cfg.namespace).Get(
+					ctx,
+					volume.PersistentVolumeClaim.ClaimName,
+					metav1.GetOptions{},
+				)
+				if err != nil {
+					return nil, errors.Wrapf(err, "error getting PVC %s", volume.PersistentVolumeClaim.ClaimName)
+				}
+				return pvc.Status.Capacity.Storage().ToDec(), nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+// contains checks if a string slice contains a given value.
+//
+// It accepts:
+// - slice: The string slice to search
+// - value: The string value to search for
+//
+// It returns:
+// - true if the slice contains the value
+// - false otherwise
+func contains(slice []string, value string) bool {
+	for _, v := range slice {
+		if v == value {
+			return true
+		}
+	}
+	return false
+}
+
+// getPercentage calculates the percentage of x out of y.
+//
+// It accepts:
+// - x: The numerator value
+// - y: The denominator value
+//
+// It returns:
+// - The percentage of x out of y, rounded to 2 decimal places.
+// - 0 if either x or y are 0 to avoid division by 0.
+func getPercentage(x, y float64) float64 {
+	if x == 0 {
+		return 0
+	}
+
+	if y == 0 {
+		return 0
+	}
+
+	return x * 100 / y
+}
+
+// getStorageUsage executes the `df -h` command inside a pod
+// to get the storage usage of a given container.
+//
+// It accepts:
+// - podName: The name of the pod
+// - namespace: The namespace of the pod
+// - containerName: The name of the container
+//
+// It returns:
+// - The storage usage of the container (or "-" if stateless)
+// - Any error that occurred while executing the command
+//
+// When the kubernetes API has a better way of getting storage usage,
+// this should be refactored to use the API.
+func getStorageUsage(podName, namespace, containerName string) (string, error) {
+	cmd := exec.Command("kubectl", "exec", "-it", podName, "-n", namespace, "--", "df", "-h")
```

Comment:
This works, but it also introduces an assumed dependency that the user has `kubectl` installed and in their path. We can use the `k8s.io/client-go/tools/remotecommand` pkg to do the same thing without assuming the user has `kubectl` at all.